### ISSUE TITLE
Invalid column from max-line-length

### ISSUE THREAD

**Issue by Arcanemagus** [stylelint/stylelint#2266]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A bug where `max-line-length` reports a column 1 past the end of the line.

> Which rule, if any, is this issue related to?

`max-line-length`

> What CSS is needed to reproduce this issue?

```scss
.random {
    // Lorem ipsum dolor sit amet. The comment Lorem ipsum dolor sit amet, consectetur adipisicing elit. Praesentium officia fugiat unde deserunt sit, tenetur! Incidunt similique blanditiis placeat ad quia possimus libero, reiciendis excepturi non esse deserunt a odit.
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "max-line-length": 100
  }
}
```

> Which version of stylelint are you using?

`7.7.1`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint bug.scss --syntax scss`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, it's related to SCSS maps.

> What did you expect to happen?

The proper column (270) to be reported.

> What actually happened (e.g. what warnings or errors you are getting)?

A column of 271, which is 1 past the end of the input is flagged.

```shell
bug.scss
 2:271  ×  Expected line length to be no more than 100 characters   max-line-length
```

A zip of everything necessary to reproduce this can be found here: [linter-stylelint_GH307.zip](https://github.com/stylelint/stylelint/files/712500/linter-stylelint_GH307.zip)

Originally filed as https://github.com/AtomLinter/linter-stylelint/issues/307.

**Comment by Arcanemagus** [comment]
@evilebottnawi This also needs the `non-standard syntax: scss` label. (I'm assuming, unless it is also showing on pure CSS...)

**Comment by evilebottnawi** [comment]
@Arcanemagus thanks :+1:

**Comment by maraisr** [comment]
I've snagged this one, will push up my PR soonish!

**Comment by jeddy3** [comment]
> I've snagged this one, will push up my PR soonish!

Is why I labelled as `wip`. Reverting to `help wanted`. Feel free to jump on it @evilebottnawi  :)

**Comment by evilebottnawi** [comment]
/cc @Arcanemagus why it should be `270`? Maybe `269`?
```shell
    // Lorem ipsum dolor sit amet. The comment Lorem ipsum dolor sit amet, consectetur adipisicing elit. Praesentium officia fugiat unde deserunt sit, tenetur! Incidunt similique blanditiis placeat ad quia possimus libero, reiciendis excepturi non esse deserunt a odit.
```
Length - 269. http://www.lettercount.com/

**Comment by davidtheclark** [comment]
No doubt this is about the way postcss-scss handles `//` comments.

**Comment by evilebottnawi** [comment]
@davidtheclark I am working on a problem I can assume it is because of `root.toString()` and it is return handled string `/* ... */`, we should check all rules where we have `root.toString()` and use `root.source.input.css`.

**Comment by evilebottnawi** [comment]
@Arcanemagus it should be `269`. not `270`, because:
```shell
.random {
    /* Lorem ipsum dolor sit amet. The comment Lorem ipsum dolor sit amet, consectetur adipisicing elit. Praesentium officia fugiat unde deserunt sit, tenetur! Incidunt similique blanditiis placeat ad quia possimus libero, reiciendis excepturi non esse deserunt a odit. */
                                                                                                                                                                                                                                                                               ^ Column report and it is 272 column
}
```
SASS:
```shell
.random {
    // Lorem ipsum dolor sit amet. The comment Lorem ipsum dolor sit amet, consectetur adipisicing elit. Praesentium officia fugiat unde deserunt sit, tenetur! Incidunt similique blanditiis placeat ad quia possimus libero, reiciendis excepturi non esse deserunt a odit.
                                                                                                                                                                                                                                                                            ^ Column report and it is 269 column (272 - 2 (`*/`) - 1 (one space before `*/`))
}
```

**Comment by evilebottnawi** [comment]
Done in https://github.com/stylelint/stylelint/pull/2340
